### ISSUE TITLE
Correct use of @static and @http for static site?

### ISSUE THREAD

**Issue by andybee** [architect/architect#495]
In my current project I'm using both pragma, using some functions for specific routes and the root route is setup to use the proxy to S3. It deploys as expected and the routes work, but I'm curious about one aspect and whether it's by design or a configuration mistake on my part.

I now have 2x paths to the static assets, I can use the S3 bucket URL, because the assets have been deployed with public access, or I can use the API Gateway URL which proxies the static files. Should the S3 bucket not be private, with the proxy having access, to avoid having 2x URLs for the same resource?

**Comment by ryanblock** [comment]
I don't think there's any configuration issue at hand. Sounds like the question is: why do we make the S3 bucket public by default?

We do that because configuring it to be private and locked down, only accessible to internal AWS resources, did not seem necessary, and allowed us to work with a single S3 configuration instead of many, conditional configs.

There's no attack vector there (that we're presently aware of) since those assets are assumed to be publicly available deploy artifacts (and you're under no obligation to share your S3 URL with anyone); on the flip side, having it public can be a helpful debugging vector.

If you thought that was too many ways to access your S3 assets, just wait until you find out about `/_static`! 🙃

**Comment by andybee** [comment]
No, that makes complete sense with the added detail. I’m sure you love reliving technical decisions over and over again! 😀
